submissions/dashakim/oop-task/js/app.js

```diff
@@ -0,0 +1,106 @@
+var gameScore = 0;
+
+const ENEMY = {
+  speed: 100,
+  varianceSpeed: 222,
+  location: [70, 155, 235]
+};
+const PLAYER = {
+  startX: 200,
+  startY: 404,
+  stepX: 101,
+  stepY: 83
+};
+
+const CANVAS = {
+  begin: -60,
+  end: 510,
+  leftSide: 0,
+  rightSide: 400,
+  top: 0,
+  bottom: 404,
+  halfCell: 50
+};
+
+var Draw = function() {};
+
+Draw.prototype.render = function() {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+var Enemy = function(x, y, speed, player) {
+  this.x = x;
+  this.y = y;
+  this.speed = speed;
+  this.player = player;
+  this.sprite = "images/car.png";
+};
+
+Enemy.prototype = Object.create(Draw.prototype);
+Enemy.prototype.constructor = Enemy;
+
+Enemy.prototype.update = function(dt) {
+  this.x += this.speed * dt;
+  if (this.x > CANVAS.end) {
+    this.x = CANVAS.begin;
+    //random speed for enemies
+    this.speed = ENEMY.speed + Math.floor(Math.random() * ENEMY.varianceSpeed);
+  }
+  // return to start position player when collide with enemy
+  if (
+    player.x > this.x - CANVAS.halfCell &&
+    player.x < this.x + CANVAS.halfCell &&
+    player.y > this.y - CANVAS.halfCell &&
+    player.y < this.y + CANVAS.halfCell
+  ) {
+    player.startPosition();
+  }
+};
+
+var Player = function() {
+  this.startPosition();
+  this.sprite = "images/char-cat-girl.png";
```

Comment:
Looks like `Player` misses props `x`, `y`. Any class prop should be either inherited or defined explicitly.
Of course in Javascript we may create arbitrary props (like `Player.startPosition()` does here) but we write code for other people to read in the first hand. In our case `Player` would inherit these from `Character`.